docs/scenarios_authoring/BasicNodes.md

```diff
@@ -157,6 +157,22 @@ UnionMemo merges multiple branches into one stream. For each incoming branch two
 }
 ```
 
+## Collector
+
+![for_each](img/collector.png)
+
+`collector` collects values from nodes that executed multiple times (e.g. for-each subsequent nodes) and store them in a list.
+
+Collector takes one argument:
+- Input expression - expression which will be collected from all nodes invocations. 
+
+For example:
+- We use `for-each` component on list `{"one", "two", "three"}`
+- Attach bellow `for-each` some node which do `.size` on each element and returns `#elementSize`
```

Comment:
1. Attach -> Connect/Add ? 
2. which do `#element.size` on each string maybe? To show more precisely final expression and to clarify that it meant to be size of those string elements - not list of strings